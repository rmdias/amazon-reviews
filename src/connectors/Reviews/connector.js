import _ from 'lodash'
import { connect } from 'react-redux'
import { reviews } from 'actions/reviews'

export default connect(
  (state, ownProps) => ({
    reviewsList: _.get(state, 'reviews.list', []),
    filters: _.get(state, 'filters', {})
  }),
  (dispatch) => ({
    onFetchReviews: page => dispatch(reviews.fetch.request(page))
  })
)
