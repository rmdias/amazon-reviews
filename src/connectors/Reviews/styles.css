.arw_reviews {
  padding-top: 30px;
}

.arw_reviews_group_type {
  padding-bottom: 10px;
}

.arw_reviews__loader {
  margin: 0 auto;
}
