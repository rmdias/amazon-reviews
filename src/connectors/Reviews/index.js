import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import { Hint } from 'luna-ui-lib'

import ReviewItem from 'components/ReviewItem'

import './styles.css'
import connector from './connector'

class Reviews extends PureComponent {
  static propTypes = {
    filters: PropTypes.object,
    reviewsList: PropTypes.array,
    onFetchReviews: PropTypes.func
  }

  componentDidMount() {
    this.props.onFetchReviews(1)
  }

  render() {
    const Items = (
      <dl className="arw_reviews_group">
        <dt className="arw_reviews_group_type"><Hint><strong>December</strong></Hint></dt>
        {this.props.reviewsList.map((review) => {
          return <ReviewItem key={review.reviewId} review={review}/>
        })}
      </dl>
    )

    return (
      <div className="arw_reviews">
        { Items }
      </div>
    )
  }
}

export default connector(Reviews)
