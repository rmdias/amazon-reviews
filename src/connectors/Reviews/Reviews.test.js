import React from 'react'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'

const middlewares = []
const mockStore = configureStore(middlewares)

import Reviews from './index'
import DEFAULT_STAR_RATING from 'constants/DEFAULT_STAR_RATING'
import ReviewsMock from './__mocks__/__mock__reviews'

import { Theme } from 'luna-ui-lib'

const filters = {
  search: '',
  groupBy: 'day',
  order: null,
  stars: [ DEFAULT_STAR_RATING ],
  loading: false
}

beforeEach(() => {
  global.scrollTo = jest.fn()
})

describe('<Reviews />', () => {
  it('toMatchSnapshot: Loading', () => {
    const initialState = { filters }
    const store = mockStore(initialState)
    const component = <Theme><Reviews store={store}/></Theme>

    const tree = renderer
      .create(component)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('toMatchSnapshot: Load <ReviewItems /> as list', () => {
    const initialState = {
      filters,
      reviews: {
        list: ReviewsMock,
        hasMore: false,
        loading: false
      }
    }
    const store = mockStore(initialState)
    const component = <Theme><Reviews store={store}/></Theme>

    const tree = renderer
      .create(component)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
