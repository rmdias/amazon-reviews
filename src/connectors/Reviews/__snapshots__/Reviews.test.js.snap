// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Reviews /> toMatchSnapshot: Load <ReviewItems /> as list 1`] = `
<div>
  <div
    className="arw_reviews"
  >
    <div>
      <dl
        className="arw_reviews_group"
      >
        <dt
          className="arw_reviews_group_type"
        >
          <span
            className="sc-dxgOiQ fWGswn"
          >
            <strong>
              07.02.2018
            </strong>
          </span>
        </dt>
        <dd>
          <div
            className="arw_reviews__item sc-ckVGcZ EUALb"
          >
            <header
              className="arw_reviews__item_header"
            >
              <img
                alt="image"
                className="arw_reviews__item_image sc-jKJlTe kkNBMm"
                src="https://images-na.ssl-images-amazon.com/images/I/undefined._SY679_.jpg"
              />
              <div
                className="arw_reviews__item_info"
              >
                <div
                  className="arw_reviews__item_info_column"
                >
                  <span
                    className="arw_reviews__item_info__title sc-eNQAEJ dWsydp"
                  >
                    date
                  </span>
                  <strong>
                    07.02.2018
                  </strong>
                </div>
                <div
                  className="arw_reviews__item_info_column"
                >
                  <span
                    className="arw_reviews__item_info__title sc-hMqMXs cMnqLP"
                  >
                    stars
                  </span>
                  <strong>
                    ★★★★★
                  </strong>
                </div>
                <div
                  className="arw_reviews__item_info_column"
                >
                  <span
                    className="arw_reviews__item_info__title sc-kEYyzF dRxZbp"
                  />
                  <strong
                    className="arw_reviews__item_name"
                  />
                </div>
              </div>
            </header>
            <h4
              className="arw_reviews__title"
            >
              <strong>
                Nice rattle sound
              </strong>
            </h4>
            <p />
          </div>
        </dd>
      </dl>
    </div>
  </div>
</div>
`;

exports[`<Reviews /> toMatchSnapshot: Loading 1`] = `
<div>
  <div
    className="arw_reviews"
  >
    <div
      className="arw_reviews__loader sc-kpOJdX jIxhBs"
      kind="primary"
      size={50}
    />
    <div />
  </div>
</div>
`;
